Specs/Core/CesiumTerrainProviderSpec.js

```diff
@@ -404,9 +404,8 @@ describe("Core/CesiumTerrainProvider", function () {
       provider.errorEvent.removeEventListener(errorMatcher);
     };
 
-    provider.errorEvent.addEventListener(errorMatcher);
-
     return provider.readyPromise.then(fail).catch((e) => {
+      provider.errorEvent.addEventListener(errorMatcher);
```

Comment:
I just realized my wording in the last comment was ambiguous-- Sorry!

Adding the event listener itself can remain where it was before. I meant that in the catch block, we need an `expect` statement to ensure that the listener was actually called. Otherwise this test could return a false positive.

Something like:

````js
const errorMatcher = function (event) {
      ...
      listenerCalled = true;
    };

    provider.errorEvent.addEventListener(errorMatcher);
    
    return provider.readyPromise.then(fail).catch((e) => {
        ...
        expect(listenerCalled).toBeTrue();
    });
    ````